Add Animation Completion Parameter

blick9 opened the pull request:
hello, I need a callback for animation when I use it by my case.
but, It was not implemented, so I made a request like this.
I hope to be approved

Commit:
Added animation completion parameter to functions that singular animate operation

Commit:
Added animation completion parameter to functions that plural animate operation

blick9 committed:
Merge branch 'master' into feature/animation_completion

blick9 committed:
Merge branch 'master' into feature/animation_completion

apang42 commented on Sources/AloeStackView/AloeStackView.swift:
```diff
@@ -551,16 +574,20 @@ open class AloeStackView: UIScrollView {
     let aboveCell = cellAbove(cell: cell)
 
-    let completion: (Bool) -> Void = { [weak self] _ in
+    let completion: (Bool) -> Void = { [weak self] flag in
```
The name of the handler conflicts with the argument to the method. Perhaps rename the local variable to `completionHandler` to disambiguate. Also the argument to the function might be better called `finished` rather than `flag`.

blick9 replied:
thank you for your feedback!
i applied all feedbacks. could you check one more time?

apang42 commented on Sources/AloeStackView/AloeStackView.swift:
```diff
@@ -569,6 +596,7 @@ open class AloeStackView: UIScrollView {
       if let aboveCell = aboveCell {
         self.updateSeparatorVisibility(forCell: aboveCell)
       }
+      completion?(flag)
```
Insert one blank line above this line.

blick9 replied:
i did

apang42 commented on Sources/AloeStackView/AloeStackView.swift:
```diff
@@ -551,16 +574,20 @@ open class AloeStackView: UIScrollView {
-      UIView.animate(withDuration: 0.3) {
+      UIView.animate(withDuration: 0.3, animations: {
         cell.alpha = 1
+      }) { flag in
```
This is a little confusing, perhaps we could split this handler out to a separate `completionHandler` above, like with the `removeCell` method?

blick9 replied:
remove block closure, just input argument

apang42 commented on Sources/AloeStackView/AloeStackView.swift:
```diff
@@ -64,35 +64,39 @@ open class AloeStackView: UIScrollView {
   /// If `animated` is `true`, the insertions are animated.
-  open func prependRows(_ rows: [UIView], animated: Bool = false) {
-    rows.reversed().forEach { prependRow($0, animated: animated) }
+  open func prependRows(_ rows: [UIView], animated: Bool = false, completion: ((Bool) -> Void)? = nil) {
```
please wrap these function signatures on to multiple lines
```
open func prependRows(
    _ rows: [UIView],
    animated: Bool = false,
    completion: ((Bool) -> Void)? = nil)
{
```

blick9 replied:
i broke the line of function name that longer then 110 characters.

blick9 replied:
however, here is my break line style.

```
open func prependRows(_ rows: [UIView],
                      animated: Bool = false,
                      completion: ((Bool) -> Void)? = nil) {
       // code is here
}
```

if you don't like this style please commit the additional feedback. i will apply as soon as possible.

apang42 commented on Sources/AloeStackView/AloeStackView.swift:
```diff
@@ -64,35 +64,39 @@ open class AloeStackView: UIScrollView {
-  open func addRow(_ row: UIView, animated: Bool = false) {
-    insertCell(withContentView: row, atIndex: stackView.arrangedSubviews.count, animated: animated)
+  open func addRow(_ row: UIView, animated: Bool = false, completion: ((Bool) -> Void)? = nil) {
+    insertCell(withContentView: row, atIndex: stackView.arrangedSubviews.count, animated: animated, completion: completion)
```
please wrap multiple statements onto separate lines
```
insertCell(
  withContentView: row, 
  atIndex: stackView.arrangedSubviews.count, 
  animated: animated, 
  completion: completion)
```

blick9 replied:
same as above comment

Commit:
Apply pull request feedbacks

- insert blank line
- change argument, variable name
- truncate line what function name that parameters length longer than 110 characters

blick9 committed:
Merge branch 'master' into feature/animation_completion

apang42 commented:
Hi @blick9, thanks for doing this work - on further consideration it feels like we might want to wait until a decision is made on https://github.com/airbnb/AloeStackView/pull/12 as it seems the implementation of custom row animations could have a big impact on this API. We'll take a look at PR#12 and get back to you on this one. Sorry for the delay on this!

blick9 commented:
Hello, it's already been a year. do you have no plans to update any more?
